File: 2020/Day_01/code.php
<?php

require_once __DIR__ . '/../../lib/php/utils.php';

$input = readInput(__DIR__);

// Task code
function part01(array $input): int
{
    $result = 0;

    foreach ($input as $firstValue) {
        foreach ($input as $secondValue) {
            $sum = $firstValue + $secondValue;
            if ($sum === 2020) {
                $result = $firstValue * $secondValue;
            }
        }
    }

    return $result;
}

function part02(array $input): int
{
    $result = 0;

    foreach ($input as $firstValue) {
        foreach ($input as $secondValue) {
            foreach ($input as $thirdValue) {
                $sum = $firstValue + $secondValue + $thirdValue;

                if ($sum === 2020) {
                    $result = $firstValue * $secondValue * $thirdValue;
                }
            }
        }
    }

    return $result;
}

// Execute
calcExecutionTime();
$result01 = part01($input);
writeln('Solution Part 1: ' . $result01);
$result02 = part02($input);
writeln('Solution Part 2: ' . $result02);
$executionTime = calcExecutionTime();

echo("Execution time: $executionTime" . PHP_EOL);
saveBenchmarkTime($executionTime, __DIR__);

// Task test
testResults(
    [935419, 49880012], // Expected
    [$result01, $result02], // Result
);

File: 2020/Day_01/code.go
package main

import (
	"advent-of-code/helper"
	"fmt"
	"path"
	"runtime"
	"strconv"
	"time"
)

func main() {
	_, f, _, _ := runtime.Caller(0)
	cwd := path.Join(path.Dir(f))

	input := helper.ReadInput(cwd, helper.NewLine)
	iValues := input.Strings()

	// Execute
	start := time.Now()
	result01 := part01(iValues)
	result02 := part02(iValues)
	executionTime := helper.ExecutionTime(time.Since(start))

	fmt.Printf("Solution Part 1: %v\n", result01)
	fmt.Printf("Solution Part 2: %v\n", result02)
	fmt.Printf("Execution time: %s\n", executionTime)

	helper.SaveBenchmarkTime(executionTime, cwd)

	// Testing
	helper.TestResults(
		[]helper.TestingValue{
			helper.TestingValue{Result: result01, Expect: 935419},
			helper.TestingValue{Result: result02, Expect: 49880012},
		},
	)
}

// Task code
func part01(input []string) int {
	for _, firstValue := range input {
		for _, secondValue := range input {
			fv, _ := strconv.Atoi(firstValue)
			sv, _ := strconv.Atoi(secondValue)

			sum := fv + sv

			if 2020 == sum {
				return fv * sv
			}
		}
	}

	return 0
}

func part02(input []string) int {
	for _, firstValue := range input {
		for _, secondValue := range input {
			for _, thirdValue := range input {
				fv, _ := strconv.Atoi(firstValue)
				sv, _ := strconv.Atoi(secondValue)
				tv, _ := strconv.Atoi(thirdValue)

				sum := fv + sv + tv

				if 2020 == sum {
					return fv * sv * tv
				}
			}
		}
	}

	return 0
}
